src/aro/Driver.zig

```diff
@@ -892,3 +916,214 @@ fn exitWithCleanup(d: *Driver, code: u8) noreturn {
+/// Then, smooshes them together with platform defaults, to decide whether
+/// this compile should be using PIC mode or not.
+/// Returns a tuple of ( backend.CodeGenOptions.PicLevel, IsPIE).
+pub fn getPICMode(d: *Driver, args: []const []const u8, lastpic: []const u8) !struct { backend.CodeGenOptions.PicLevel, bool } {
+    const eqlIgnoreCase = std.ascii.eqlIgnoreCase;
+
+    const target = d.comp.target;
+    const linker = d.use_linker orelse @import("system_defaults").linker;
+    const is_bfd_linker = eqlIgnoreCase(linker, "bfd");
+
+    const is_pie_default = switch (target_util.isPIEDefault(target)) {
+        .yes => true,
+        .no => false,
+        .depends_on_linker => if (is_bfd_linker)
+            target.cpu.arch == .x86_64 // CrossWindows
+        else
+            false, //MSVC
+    };
+    const is_pic_default = switch (target_util.isPICdefault(target)) {
+        .yes => true,
+        .no => false,
+        .depends_on_linker => if (is_bfd_linker)
+            target.cpu.arch == .x86_64
+        else
+            (target.cpu.arch == .x86_64 or target.cpu.arch == .aarch64),
+    };
+
+    var pie: bool = is_pie_default;
+    var pic: bool = pie or is_pic_default;
+    // The Darwin/MachO default to use PIC does not apply when using -static.
+    if (target.ofmt == .macho and d.static) {
+        pic, pie = .{ false, false };
+    }
+    var is_piclevel_two = pic;
+
+    const kernel_or_kext: bool = d.mkernel or d.apple_kext;
+
+    // Android-specific defaults for PIC/PIE
+    if (target.isAndroid()) {
+        switch (target.cpu.arch) {
+            .arm,
+            .armeb,
+            .thumb,
+            .thumbeb,
+            .aarch64,
+            .mips,
+            .mipsel,
+            .mips64,
+            .mips64el,
+            => pic = true, // "-fpic"
+
+            .x86, .x86_64 => {
+                pic = true; // "-fPIC"
+                is_piclevel_two = true;
+            },
+            else => {},
+        }
+    }
+
+    // OHOS-specific defaults for PIC/PIE
+    if (target.abi == .ohos and target.cpu.arch == .aarch64)
+        pic = true;
+
+    // OpenBSD-specific defaults for PIE
+    if (target.os.tag == .openbsd) {
+        switch (target.cpu.arch) {
+            .arm, .aarch64, .mips64, .mips64el, .x86, .x86_64 => is_piclevel_two = false, // "-fpie"
+            .powerpc, .sparc64 => is_piclevel_two = true, // "-fPIE"
+            else => {},
+        }
+    }
+
+    // The last argument relating to either PIC or PIE wins, and no
+    // other argument is used. If the last argument is any flavor of the
+    // '-fno-...' arguments, both PIC and PIE are disabled. Any PIE
+    // option implicitly enables PIC at the same level.
+    if (target.os.tag == .windows and
+        !target_util.isCygwinMinGW(target) and
+        // d.lastpic_pos != null and
+        StaticStringSet.initComptime(.{ .{"-fPIC"}, .{"-fpic"}, .{"-fpie"}, .{"-fPIE"} }).has(lastpic))
```

Comment:
Now that `lastpic` arg is checked separately `std.ascii.eqlIgnoreCase` could work nicely for these.